assets/javascripts/discourse/components/topic-chat-float.js

```diff
@@ -129,7 +129,15 @@ export default Component.extend({
       ajax(`/chat/chat_channels/${channel.id}/follow`, { method: "POST" })
         .then(() => {
           this.chat.startTrackingChannel(channel);
-          this.switchChannel(channel);
+          if (this.site.mobileView) {
+            return this.router.transitionTo(
```

Comment:
Could it make to move this logic into `this.switchChannel`?

Reply:
yeah we probably should, good idea.